### PR TITLE
Round trip with SSDT

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR implements the pieces required for a project created in SSDT to open and build without any manual intervention from the user.
A project created in SSDT doesn't have the NetCore build information and reference assembly information in its sqlproj, which is required for Build to work in ADS. 
This would allow a smooth round tripping of database projects between ADS and SSDT.

I will add related tests in another PR.

### CODE REVIEW

File: extensions/sql-database-projects/src/models/project.ts [hunk 1]
```diff
@@ -51,6 +52,47 @@ export class Project {
...
+	private async updateImportToSupportRountTrip(): Promise<void> {
+
+		for (let ig = 0; ig < this.projFileXmlDoc.documentElement.getElementsByTagName(constants.Import).length; ig++) {
+			const importTarget = this.projFileXmlDoc.documentElement.getElementsByTagName(constants.Import)[ig];
+
+			let condition = importTarget.getAttribute(constants.Condition);
+			let project = importTarget.getAttribute(constants.Project);
+
+			if (condition === constants.SqlDbPresentCondition && project === constants.SqlDbTargets) {
+				await this.updateImportedTargetsToProjFile(constants.RoundTripSqlDbPresentCondition, project, importTarget);
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/project.ts:86]: add comments about what's being updated

File: extensions/sql-database-projects/src/models/project.ts [hunk 2]
```diff
@@ -127,6 +169,37 @@ export class Project {
...
+		else {
+			this.projFileXmlDoc.documentElement.appendChild(importNode, oldImportNode);
+		}
+
+		await this.serializeToProjFile(this.projFileXmlDoc);
+		return importNode;
+	}
+
+	private async updatePackageReferenceInProjFile(): Promise<void> {
+		let outputItemGroup = this.projFileXmlDoc.createElement(constants.ItemGroup);
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/models/project.ts:189]: do we see ref in tree now?

Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/models/project.ts:189]: No.

File: extensions/sql-database-projects/src/models/project.ts [hunk 3]
```diff
@@ -51,6 +52,47 @@ export class Project {
 				this.files.push(this.createProjectEntry(itemGroup.getElementsByTagName(constants.Folder)[f].getAttribute(constants.Include), EntryType.Folder));
 			}
 		}
+
+		// find all import statements to include
+
+		for (let ig = 0; ig < this.projFileXmlDoc.documentElement.getElementsByTagName(constants.Import).length; ig++) {
```
[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/models/project.ts:58]: My bad. "ig" at another place was to signify ItemGroup, but it doesn't go well with Import here. Updating it appropriately.

[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/models/project.ts]: Updated it with "i" for "Import"

File: extensions/sql-database-projects/src/common/constants.ts [hunk 4]
```diff
@@ -57,6 +59,7 @@ export const projectNameRequired = localize('projectNameRequired', "Name is requ
 export const projectLocationRequired = localize('projectLocationRequired', "Location is required to create a new database project.");
 export const schemaCompareNotInstalled = localize('schemaCompareNotInstalled', "Schema compare extension installation is required to run schema compare");
 export const buildDacpacNotFound = localize('buildDacpacNotFound', "Dacpac created from build not found");
+export const updateProjectForRoundTrip = localize('updateProjectForRoundTrip', "To build this project, ADS needs to update targets and references. If the project is created in SSDT, it will continue to work in both tools. Do you want ADS to update the project?");
```
Comment (Benjin) [inline extensions/sql-database-projects/src/common/constants.ts:62]: "Azure Data Studio" instead of "ADS"

**Pull request merged**